### ISSUE TITLE
Support for gsync/freesync

### ISSUE THREAD

**Issue by OopyDoopy** [jellyfin/jellyfin-media-player#91]
This is something that MPV Shim already supports.  Since this application uses d3d11 for rendering (same as MPV Shim), supporting variable refresh rate screens should be simply switching to the dxgi flip model, then the graphics driver handles the rest.  Unfortunately, that's where my knowledge on the matter ends.

This is the best solution to mismatched refresh rates possible on PC, provided the user has the hardware (and you basically can't buy a monitor that doesn't support it these days).

**Comment by Chickenbreadlp** [comment]
It looks like JF Media Player already supports VRR when enabling OpenGL. I certainly noticed mouse cursor movement was a lot less smooth with OpenGL, like as if the monitor was running at 24Hz or smth. This unfortunately also turns off HDR when playing back in Fullscreen :(

**Comment by sevenrats** [comment]
what is the state of this issue guys? id like to close it because it seems resolved.

**Comment by Chickenbreadlp** [comment]
I honestly can't really tell if JF Media Player v1.9.1 is making use of VRR or not.
There is a setting for automatically setting the Refreshrate, but it seems to actively change the Windows refreshrate, rather then use VRR, besides it only works when running the entire client in Fullscreen, and not just enable FS during playback.
This setting does set the refreshrate to the highest integer doubling of the contents framerate that the display can support (say 100Hz for 25Hz content on a 120Hz screen). As such, if it were to use VRR in the same way, then it'd be hard to tell given the relatively minor difference between 120Hz and 100Hz...

As I had originally mentioned, when using OpenGL it definitely uses VRR to go down the framerate of the content, and not some frame doubled counterpart, as cursor movement get's rather difficult, but the OpenGL backend does not support HDR.

**Comment by iwa** [comment]
would really love to see this implemented someday, it's really awful to use when watching content

**Comment by nalabrie** [comment]
Just chiming in to mention that the OpenGL setting does enable VRR for me as well. Which is interesting because in regular MPV using OpenGL (or any GPU decoder) disables VRR. After some brief searching online, that issue with MPV _might_ be more related to a Nvidia driver bug and I certainly didn't fiddle with all available decoders. Sorry for going off topic, it is just curious that Jellyfin Media Player seems to work when it uses MPV underneath and I wonder what is different exactly.

**Comment by iwa** [comment]
@andrewrabert you mean fixed in the pre-release/nightly?

**Comment by andrewrabert** [comment]
@iwa yup. Haven't tested myself, but I'd be happy to reopen this if it's still an issue.
